### PR TITLE
feat: update device admin state

### PR DESCRIPTION
**Pull request by difince**
Enable update of device admin state by deviceName or deviceId
Usage Examples:
./edgex-cli device adminstate update  -s LOCKED -d TestRandom-Integer-Device
./edgex-cli device adminstate update  -s unlocked --deviceId 089d3bbf-3795-4ac6-b471-e9645969f720

Closes: https://github.com/edgexfoundry/edgex-cli/issues/247

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/device/adminstate/update/update.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// NewCommand returns the update deviceName command
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update deviceName admin state",
+		Long:  `Update deviceName admin state`,
+		RunE:  updateAdminStatusHandler,
+	}
+	cmd.Flags().StringVarP(&state, "state", "s", "", fmt.Sprintf("Admin State \nValues: [%s]", strings.Join([]string{models.Locked, models.Unlocked}, ",")))
+	cmd.Flags().StringVarP(&deviceName, "deviceName", "d", "", "Name of the deviceName to be updated")
```
Comment (mkbhanda) [inline cmd/device/adminstate/update/update.go:31]: Would it have been just as clear to have flags such as 
n "name".   and
i "id"

**Review by mkbhanda — COMMENTED** [review]
Shall merge if you think renaming flags not particularly clear or useful.

File: cmd/device/adminstate/update/update.go (hunk 1 above)
Comment (difince) [reply cmd/device/adminstate/update/update.go:31]: Agree!

Comment (difince) [reply cmd/device/adminstate/update/update.go]: Resolved

File: cmd/device/adminstate/update/update.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// NewCommand returns the update deviceName command
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update deviceName admin state",
+		Long:  `Update deviceName admin state`,
+		RunE:  updateAdminStatusHandler,
+	}
+	cmd.Flags().StringVarP(&state, "state", "s", "", fmt.Sprintf("Admin State \nValues: [%s]", strings.Join([]string{models.Locked, models.Unlocked}, ",")))
+	cmd.Flags().StringVarP(&deviceName, "name", "d", "", "Name of the deviceName to be updated")
```
Comment (mkbhanda) [inline cmd/device/adminstate/update/update.go:31]: s/"d"/"n".

**Review by mkbhanda — COMMENTED** [review]
a nit, sorry

File: cmd/device/adminstate/update/update.go (hunk 2 above)
Comment (difince) [reply cmd/device/adminstate/update/update.go]: fixed!

File: cmd/device/adminstate/update/update.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+func validate() error {
+	if state == "" {
+		return errors.New("admin state should be specified")
+	} else if state != "" {
+		valid, err := models.AdminState(state).Validate()
+		if !valid {
+			return err
+		}
+	} else if deviceName == "" && deviceId == "" {
```
Comment (mkbhanda) [inline cmd/device/adminstate/update/update.go:62]: s/pe/be

File: cmd/device/adminstate/update/update.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return errors.New("deviceName or deviceId should pe specified")
+	} else if deviceName != "" && deviceId != "" {
+		return errors.New("only one deviceName or deviceId should pe specified")
```
Comment (mkbhanda) [inline cmd/device/adminstate/update/update.go:65]: s/pe/be

**Review by mkbhanda — COMMENTED** [review]
another nit, I am sorry for not catching earlier.

File: cmd/device/adminstate/update/update.go (hunk 4 above)
Comment (difince) [reply cmd/device/adminstate/update/update.go]: fixed

File: cmd/device/adminstate/update/update.go (hunk 3 above)
Comment (difince) [reply cmd/device/adminstate/update/update.go:62]: fixed

**Comment by difince** [comment]
@mkbhanda Thank you for reviewing

**Review by mkbhanda — APPROVED** [review]

**Pull request merged**